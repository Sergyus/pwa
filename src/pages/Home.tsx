import Title from '../components/Title';
import pic from './../assets/img/qwe.jpg';

export default function Home(): JSX.Element {
  return (
    <>
      <Title title="Home" />
      <img src={pic} alt="pic" />
    </>
  );
}
