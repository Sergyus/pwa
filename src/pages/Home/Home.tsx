import React from 'react';
import Title from '@components/Title';
// import pic from '@assets/img/qwe.jpg';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import { useTranslation } from 'react-i18next';
import styles from './home.module.css';

export default function Home(): JSX.Element {
  const { t } = useTranslation(['menu']);

  return (
    <div className={styles.homePage}>
      <Title title={t('home')} />
      <h1>{t('translation:React')}</h1>
      {/*<img src={pic} alt="pic" />*/}

      <br />
      <br />

      <Card>
        <CardContent>
          <Typography gutterBottom variant="h6" component="h2">
            WELCOME
          </Typography>
          <Typography variant="body2" color="textSecondary" component="p">
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Alias
            asperiores, aut beatae blanditiis cupiditate dolore dolorum ea esse
            eum explicabo harum incidunt laudantium maxime nam necessitatibus
            nobis non obcaecati odio porro possimus quam, quisquam ratione
            repudiandae tempora tenetur ullam voluptatibus voluptatum. Ab
            accusamus aliquid asperiores beatae consequatur consequuntur culpa
            cumque delectus doloremque enim excepturi facilis, harum ipsam,
            maiores minima minus molestiae numquam odio optio placeat quaerat
            quo quod quos rem sed sequi tempora ut? Asperiores eius eum facere
            fugiat, illo incidunt, ipsa laborum laudantium magnam maxime minima
            necessitatibus nihil nisi quis suscipit temporibus voluptatum. Ab ad
            autem beatae, commodi consectetur corporis cumque cupiditate
            deserunt distinctio ea eligendi esse est ex facilis inventore ipsa
            laboriosam laudantium magnam maxime molestiae nesciunt nisi odit
            officiis praesentium quaerat quasi quis ratione reprehenderit rerum
            vel veniam voluptas voluptate voluptatem? Debitis dignissimos
            doloremque eos error et eveniet iure iusto maiores neque! Aspernatur
            eaque enim natus qui.
          </Typography>
          <br />
          <Typography variant="body2" color="textSecondary" component="p">
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Alias
            asperiores, aut beatae blanditiis cupiditate dolore dolorum ea esse
            eum explicabo harum incidunt laudantium maxime nam necessitatibus
            nobis non obcaecati odio porro possimus quam, quisquam ratione
            repudiandae tempora tenetur ullam voluptatibus voluptatum. Ab
            accusamus aliquid asperiores beatae consequatur consequuntur culpa
            cumque delectus doloremque enim excepturi facilis, harum ipsam,
            maiores minima minus molestiae numquam odio optio placeat quaerat
            quo quod quos rem sed sequi tempora ut? Asperiores eius eum facere
            fugiat, illo incidunt, ipsa laborum laudantium magnam maxime minima
            necessitatibus nihil nisi quis suscipit temporibus voluptatum. Ab ad
            autem beatae, commodi consectetur corporis cumque cupiditate
            deserunt distinctio ea eligendi esse est ex facilis inventore ipsa
            laboriosam laudantium magnam maxime molestiae nesciunt nisi odit
            officiis praesentium quaerat quasi quis ratione reprehenderit rerum
            vel veniam voluptas voluptate voluptatem? Debitis dignissimos
            doloremque eos error et eveniet iure iusto maiores neque! Aspernatur
            eaque enim natus qui.
          </Typography>
        </CardContent>
      </Card>
    </div>
  );
}
