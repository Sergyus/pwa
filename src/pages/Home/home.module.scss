.home-page {
  margin-bottom: 10px;
}

.test-modules {
  display: flex;
}
.Home {
}
