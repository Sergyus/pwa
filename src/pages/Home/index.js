/* eslint-disable @typescript-eslint/explicit-module-boundary-types */

/*
 * Due to this known issue: https://github.com/smooth-code/loadable-components/issues/173
 * Use .js extension for code-splitting file
 */

import React from 'react';
import loadable from '@loadable/component';
import Loading from '../../components/Loading';

const Home = loadable(() => import('./Home'), {
  // eslint-disable-next-line react/jsx-filename-extension
  fallback: <Loading />,
});

// eslint-disable-next-line react/display-name
export default (props) => <Home {...props} />;
