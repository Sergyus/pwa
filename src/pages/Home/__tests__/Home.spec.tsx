import { render } from '@testing-library/react';
import React from 'react';
import Home from '../Home';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

describe('<Home />', () => {
  it('should renders', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });
});
