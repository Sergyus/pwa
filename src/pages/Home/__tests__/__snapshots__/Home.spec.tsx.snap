// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should renders 1`] = `
<DocumentFragment>
  <div
    class="HomePage"
  >
    <h1
      class="title"
    >
      home
    </h1>
    <h1>
      translation:React
    </h1>
    <img
      alt="pic"
      src="IMAGE_MOCK"
    />
    <br />
    <br />
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
        >
          WELCOME
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Alias asperiores, aut beatae blanditiis cupiditate dolore dolorum ea esse eum explicabo harum incidunt laudantium maxime nam necessitatibus nobis non obcaecati odio porro possimus quam, quisquam ratione repudiandae tempora tenetur ullam voluptatibus voluptatum. Ab accusamus aliquid asperiores beatae consequatur consequuntur culpa cumque delectus doloremque enim excepturi facilis, harum ipsam, maiores minima minus molestiae numquam odio optio placeat quaerat quo quod quos rem sed sequi tempora ut? Asperiores eius eum facere fugiat, illo incidunt, ipsa laborum laudantium magnam maxime minima necessitatibus nihil nisi quis suscipit temporibus voluptatum. Ab ad autem beatae, commodi consectetur corporis cumque cupiditate deserunt distinctio ea eligendi esse est ex facilis inventore ipsa laboriosam laudantium magnam maxime molestiae nesciunt nisi odit officiis praesentium quaerat quasi quis ratione reprehenderit rerum vel veniam voluptas voluptate voluptatem? Debitis dignissimos doloremque eos error et eveniet iure iusto maiores neque! Aspernatur eaque enim natus qui.
        </p>
        <br />
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Alias asperiores, aut beatae blanditiis cupiditate dolore dolorum ea esse eum explicabo harum incidunt laudantium maxime nam necessitatibus nobis non obcaecati odio porro possimus quam, quisquam ratione repudiandae tempora tenetur ullam voluptatibus voluptatum. Ab accusamus aliquid asperiores beatae consequatur consequuntur culpa cumque delectus doloremque enim excepturi facilis, harum ipsam, maiores minima minus molestiae numquam odio optio placeat quaerat quo quod quos rem sed sequi tempora ut? Asperiores eius eum facere fugiat, illo incidunt, ipsa laborum laudantium magnam maxime minima necessitatibus nihil nisi quis suscipit temporibus voluptatum. Ab ad autem beatae, commodi consectetur corporis cumque cupiditate deserunt distinctio ea eligendi esse est ex facilis inventore ipsa laboriosam laudantium magnam maxime molestiae nesciunt nisi odit officiis praesentium quaerat quasi quis ratione reprehenderit rerum vel veniam voluptas voluptate voluptatem? Debitis dignissimos doloremque eos error et eveniet iure iusto maiores neque! Aspernatur eaque enim natus qui.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
